contributors/design-proposals/autoscaling/hpa-step-scaling.md

```diff
@@ -0,0 +1,143 @@
+# HPA: Threshold Based Step Scaling Policy Proposal
+
+## Target of HPA
+
+The target of HPA is to dynamically adjusts the amount of allocated resources (pods) to meet changes in
+workload demands. Basically, we evaluate a good HPA scaling policy as:
+
+* Can achieve high resource utilization
+* Responsive: scale to the desire number of replicas fast
+
+## Problem
+
+The Horizontal Pod Autoscaler (HPA) is implemented as a Kubernetes API resource and a controller.
+Current HPA only implement “target tracking based scaling policy”: the controller periodically adjusts 
+the number of replicas in a replication controller or deployment to match the observed average CPU 
+utilization to the target specified by user.
+
+In a nutshell, the target number of pods is calculated from the following formula:
+```
+TargetNumOfPods = ceil(sum(CurrentUtilization) / TargetUtilization)
+```
+
+Howeever, this kind of “target tracking based scaling policy” does NOT perform well when we want to maintain high 
+resource utilization as well as want the HPA to scale fast, especailly when the workload deamds has spikes. 
+To solve this problem, we propose to add a “threshold based step scaling policy” to HPA. We will discuss the
+mechainsm of “threshold based step scaling policy” and the scenarios where it performs better than current
+“target tracking based scaling policy” in the next two sections.
+
+
+## Threshold Based Step Scaling
+
+With step scaling policy, you can add one or more step adjustments that enable you to scale based on the
+threshold breach. Each step adjustment specifies a lower bound for the metric value, an upper bound for 
+the metric value, and the amount by which to scale, based on the scaling adjustment type. 
+
+The HPA controller periodically pull utilization information, the it compares the CurrentUtilization 
+against the upper and lower bounds defined by the step adjustments to determine which step adjustment 
+to perform.
+
+A concrete example for step scaling policy is:
+
+* Lower bound: NULL, Upper bound: 20, Adjustment: -2; 
+* Lower bound: 20, Upper bound: 40, Adjustment: -1;
+* Lower bound: 60, Upper bound: 80, Adjustment: +1;
+* Lower bound: 80, Upper bound: NULL, Adjustment: +2;
```

Comment:
You should provide examples of HPA yaml to clarify spec change. 

Just my thoughts: it would be tricky if user need to configure  complex policies like those above, and also, lack of flexibility when pod changes or policy updated. I would prefer to see sth more general than hard coded boundaries: they are not so straightforward to most users.